lib/js-file.js

```diff
@@ -308,50 +359,67 @@ JsFile.prototype = {
     /**
      * Returns the first token for the file.
      *
+     * @param {Option} [options]
+     * @param {Boolean} [options.includeComments=false]
+     * @param {Boolean} [options.includeWhitespace=false]
      * @returns {Object}
      */
-    getFirstToken: function() {
-        return this._tokens[0];
+    getFirstToken: function(options) {
+        return this._getTokenFromIndex(0, 1, options);
     },
 
     /**
      * Returns the last token for the file.
      *
+     * @param {Option} [options]
+     * @param {Boolean} [options.includeComments=false]
+     * @param {Boolean} [options.includeWhitespace=false]
      * @returns {Object}
      */
-    getLastToken: function() {
-        return this._tokens[this._tokens.length - 1];
+    getLastToken: function(options) {
+        return this._getTokenFromIndex(this._tokens.length - 1, -1, options);
     },
 
     /**
-     * Returns the first token before the given.
+     * Returns the first token after the given using direction and specified conditions.
      *
-     * @param {Object} token
+     * @param {Number} index
+     * @param {Number} direction `1` - forward or `-1` - backwards
      * @param {Object} [options]
      * @param {Boolean} [options.includeComments=false]
+     * @param {Boolean} [options.includeWhitespace=false]
      * @returns {Object|null}
      */
-    getPrevToken: function(token, options) {
-        var index = token._tokenIndex - 1;
-        if (index < 0) {
-            return null;
-        }
-
-        if (options && options.includeComments) {
-            return this._tokens[index] || null;
-        }
+    _getTokenFromIndex: function(index, direction, options) {
```

Comment:
Not a big deal but I don't really like the `direction `1` - forward or `-1` - backwards`
since you can't tell what it means from looking at `this._getTokenFromIndex(this._tokens.length - 1, -1, options);` Oh but I see it's an internal method anyway